src/components/graph/network-modification-tree-model.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import { getLayoutedElements } from './layout';
+
+export default class NetworkModificationTreeModel {
+    treeElements = [];
+
+    updateLayout() {
+        this.treeElements = getLayoutedElements(this.treeElements);
+    }
+
+    addChild(element, parentId) {
+        // Add node
+        this.treeElements.push({
```

Comment:
We don't check if the element already exist ?

Reply:
We could, but this might introduce a performance penalty. I prefer not to check it for now.